src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -0,0 +1,218 @@
+        }
+        else
+        {
+            winrt::com_ptr<IUIAutomation> automation;
+            THROW_IF_FAILED(::CoCreateInstance(__uuidof(CUIAutomation8), nullptr, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(automation.put())));
+
+            winrt::com_ptr<IUIAutomationCacheRequest> cacheRequest;
+            automation->CreateCacheRequest(cacheRequest.put());
+            return cacheRequest;
+        }
+    }
+
+    TEST_CLASS(UiaOperationAbstractionTests)
+    {
+    public:
+        // Asserts that you can get the name of a UiaElement.
+        void ElementGetNameTest(const bool useRemoteOperations)
+        {
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+            auto calc = WaitForElementFocus(L"Display is 0");
+
+            winrt::com_ptr<IUIAutomation> automation;
+            THROW_IF_FAILED(::CoCreateInstance(__uuidof(CUIAutomation8), nullptr, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(automation.put())));
+            UiaOperationAbstraction::Initialize(useRemoteOperations, automation.get());
+            auto cleanup = wil::scope_exit([&]()
+            {
+                UiaOperationAbstraction::Cleanup();
+            });
+
+            auto scope = UiaOperationScope::StartNew();
+
+            UiaElement element = calc;
+            auto name = element.GetName(false /*useCachedApi*/);
+            scope.BindResult(name);
+
+            scope.Resolve();
+
+            Assert::AreEqual(std::wstring(static_cast<wil::shared_bstr>(name).get()), std::wstring(L"Display is 0"));
+        }
+        TEST_METHOD(ElementGetNameLocalTest)
+        {
+            GetParentElementTest(false);
+        }
+
+        TEST_METHOD(ElementGetNameRemoteTest)
+        {
+            GetParentElementTest(true);
+        }
+
+        // Asserts that GetEnclosingElement can be called with a cache request.
+        void GetEnclosingElementTest(const bool useRemoteOperations)
+        {
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+            auto calc = WaitForElementFocus(L"Display is 0");
+
+            winrt::com_ptr<IUIAutomation> automation;
+            THROW_IF_FAILED(::CoCreateInstance(__uuidof(CUIAutomation8), nullptr, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(automation.put())));
+            UiaOperationAbstraction::Initialize(useRemoteOperations, automation.get());
+            auto cleanup = wil::scope_exit([&]()
+            {
+                UiaOperationAbstraction::Cleanup();
+            });
+
+            auto scope = UiaOperationScope::StartNew();
+
+            UiaElement element = calc;
+            UiaTextPattern textPattern = element.GetTextPattern(false /*useCached*/);
+            UiaTextRange textRange = textPattern.GetDocumentRange();
+
+            auto cacheRequest = CreateCacheRequest(useRemoteOperations);
+            cacheRequest.AddProperty(static_cast<winrt::AutomationPropertyId>(UIA_NamePropertyId));
```

Comment:
If possible, I would add a mirroring test for retrieving and using cached patterns.

Reply:
Never mind. It's in the other test suite.

---
In reply to: [459171231](https://github.com/microsoft/Microsoft-UI-UIAutomation/pull/10#discussion_r459171231) [](ancestors = 459171231)

Reply:
Fair. It wouldn't hurt to have extra coverage on that.

Reply:
That covers the other of the 2 functionalities behind this new feature.

And you have most of the pieces here already (`Text` pattern + getting the document range + getting text + comparing the text would do just fine for the test imho).

---
In reply to: [459176822](https://github.com/microsoft/Microsoft-UI-UIAutomation/pull/10#discussion_r459176822) [](ancestors = 459176822)